### PR TITLE
Issues #398: tenant_show_new, add/remove content admin

### PR DESCRIPTION
**Pull request by elv-Noppapon**
**Organizing OG Tenant**

**TenantShowNew**: The original **TenantShow**- [link](https://github.com/elv-Noppapon/elv-live-js/blob/87c84faab4e543638a6d3d69ca84adee4a8db4b8/src/EluvioLive.js#L180) is expected to be renamed to something along the line of TenantShowNft and replaced by this new function. The function returns the tenant's admins group and content's admins by calling the GroupsMapping field on the tenant's contract.

**AddContentAdmin** and **RemoveContentAdmin**: Given 1.) tenantId and 2.) contentAdminsAddress, the function make corresponding addition/removal to the GroupsMapping["content_admin"] field on the contract.

**CLI/Function Exposure Notes**: All of the above functions can be called directly from the CLI. However, I noticed that similar functions such as [TenantAddNft](https://github.com/elv-Noppapon/elv-live-js/blob/87c84faab4e543638a6d3d69ca84adee4a8db4b8/src/EluvioLive.js#L509) and TenantRemoveNft are not exposed to the CLI directly, but are part of other command lines/functions. We might want something similar here?

**Testing**: 
Screen Shot 2023-06-15 at 10 51 07 AM

Screen Shot 2023-06-15 at 10 52 24 AM

TenantShowNew behaves expectedly, but AddContentAdmin and RemoveContentAdmin (called from the command line) do not successfully update the contract. I believe this might be because the `onlyAdmin` tag on the `addGroup`/`removeGroup` functions. I would appreciate any input here!

**June 19 Updates**

**TenantShowNew** now returns a warning message if tenant doesn't have content admin's group in GroupsMapping.
**tenant_fix** running tenant_fix as owner will create a new content admin's group where the owner is the group's manager.
**TenantCreate/Deploy** creates/adds content admin's group (initially only tenant admin's group).

**Testing**:
space_tenant_create -> tenant_show_new -> tenant_remove_content_admin -> tenant_show_new -> tenant_add_content_admin -> tenant_show_new behaves as expected.

### CODE REVIEW

File: src/EluvioLive.js
Comment (elv-nate) [inline src/EluvioLive.js:420]: This log can be removed

File: src/EluvioLive.js
Comment (elv-nate) [inline src/EluvioLive.js:445]: Also this one

**Review by elv-nate — COMMENTED** [review]
Looking at it briefly, it all seems good to me. Some small nits around print statements. 

Re: CLI exposure, I think it makes sense to have this functionality exposed on a CLI.

Definitely get the add/remove functions to pass somehow before merging though. I don't see anything wrong with the key, so most likely the problem is the `PRIVATE_KEY` environment variable not having admin permissions to the tenant or something. I'm not too familiar with it.

File: src/EluvioLive.js
Comment (elv-Noppapon) [reply src/EluvioLive.js:420]: Thank you! I had them for testing but forgot to remove.

**Comment by elv-Noppapon** [comment]
> Looking at it briefly, it all seems good to me. Some small nits around print statements.
> 
> Re: CLI exposure, I think it makes sense to have this functionality exposed on a CLI.
> 
> Definitely get the add/remove functions to pass somehow before merging though. I don't see anything wrong with the key, so most likely the problem is the `PRIVATE_KEY` environment variable not having admin permissions to the tenant or something. I'm not too familiar with it.

You're right, the problem was with not having admin permissions. Resolved, thanks!

File: src/EluvioLive.js
Comment (elv-Noppapon) [inline src/EluvioLive.js:432]: Should KMS address be added to content admin group here?

File: src/EluvioLive.js
Comment (elv-Noppapon) [inline src/EluvioLive.js:455]: Call cannot override gasLimit error is thrown when the content admin group doesn't exist. I am wondering if there is better way to check this.

**Review by elv-serban — COMMENTED** [review]
### When creating a tenant:

- we create the Tenant Admins and Content Admins group
- also add _ELV_TENANT_ID to both these contracts (such that we can later retrieve the tenant ID given the group address

In tenant_show, verify that both groups have the _ELV_TENANT_ID and that it is set correctly 


### In `elv-client-js`:

Add a function DisambiguateTenantId(string) that takes either the tenant admins group (in iten format) or the actual tenant contract Id (in iten format).
- find out if this is a group or a tenant contract  (potentially by checking the presence of a method - I don't know if there are better ways to check)
- if it is a group, then read the _ELV_TENANT_ID of the group to retrieve the tenant contract Id
- return an object
```
  'tenantAdminsGroupId': '',
  'tenantId': ''
```

The idea is that given an Id that is either the tenant contract or the tenant admins group ID, figure out the tenant info.

File: src/EluvioLive.js
Comment (elv-serban) [inline src/EluvioLive.js:394]: Similarly content_admins_address,  contentAdminsAddr

File: src/ElvSpace.js
Comment (elv-serban) [inline src/ElvSpace.js:109]: Here we should also add the _ELV_TENANT_ID contract metadata to the tenant admins and content admins groups.

File: src/ElvSpace.js
Comment (elv-serban) [inline src/ElvSpace.js:186]: We should call this 'content_admins' (plural).  Is the tenant_admins currently also plural.

File: src/Provision.js
Comment (elv-serban) [inline src/Provision.js:86]: I think we should retrieve the tenant_admins group from the contract just like the content admins below.

We currently read it from userProfileClient but we should not rely on that (because we want to deprecate that feature)

File: utilities/EluvioCli.js
Comment (elv-serban) [inline utilities/EluvioCli.js:212]: Same observation here would like to use plural for the groups  'tenant_admins' 'content_admins'

File: src/ElvSpace.js
Comment (elv-Noppapon) [reply src/ElvSpace.js:186]: The tenant admin group on the tenant contract has always been referred as `"tenant_admin"` (without an s), so I thought it would be the best to keep it consistent. Changing it to `"tenant_admins"` would require changes on both the contract and the client + deployed contracts will break so we also need to create a method that fix them.

File: utilities/EluvioLiveCli.js
Comment (elv-Noppapon) [reply utilities/EluvioLiveCli.js:2197]: I am thinking it might be good to change every instance of `(content/tenant) admins address` that refers to a group to `(content/tenant) admin group address` - might be more clear this way, because it is the groups that are located in these addresses.

Comment (elv-Noppapon) [reply utilities/EluvioLiveCli.js:2197]: Also, I noticed there are other times where the code doesn't respect plurality, for example: [link](https://github.com/eluv-io/elv-live-js/blob/b1d11893bcd11af1b8599e6860b64e0b8dea7aea/src/EluvioLive.js#L77-L86). I think it's worth having a clean-up ticket, which can be assigned to me, or another PR to fix this.

**Comment by elv-Noppapon** [comment]
> ### When creating a tenant:
> * we create the Tenant Admins and Content Admins group
> * also add _ELV_TENANT_ID to both these contracts (such that we can later retrieve the tenant ID given the group address
> 
> In tenant_show, verify that both groups have the _ELV_TENANT_ID and that it is set correctly
> 
> ### In `elv-client-js`:
> Add a function DisambiguateTenantId(string) that takes either the tenant admins group (in iten format) or the actual tenant contract Id (in iten format).
> 
> * find out if this is a group or a tenant contract  (potentially by checking the presence of a method - I don't know if there are better ways to check)
> * if it is a group, then read the _ELV_TENANT_ID of the group to retrieve the tenant contract Id
> * return an object
> 
> ```
>   'tenantAdminsGroupId': '',
>   'tenantId': ''
> ```
> 
> The idea is that given an Id that is either the tenant contract or the tenant admins group ID, figure out the tenant info.

Thank you! Seems like there is already a way to DisambiguateTenantId(string) by calling this.client.authClient.AccessType(id). It returns either 'tenant' or 'group'.

File: utilities/EluvioCli.js
Comment (elv-serban) [inline utilities/EluvioCli.js:903]: Let's keep one of these - `account_set_tenant` and set both the tenant admins group and tenant contract ID.  We never need to set these independently

File: src/ElvAccount.js
Comment (elv-serban) [inline src/ElvAccount.js:60]: I think here we need to use elv-client-js address to ID (we need to get the itenxxx format)

File: utilities/EluvioCli.js
Comment (elv-serban) [inline utilities/EluvioCli.js:852]: We typically use `tenant_id` but in this case we should just say `tenant` since it's simpler as CLI documentation.

File: src/ElvAccount.js
Comment (elv-Noppapon) [reply src/ElvAccount.js]: Thanks Serban! I think the current approach is correct, as we are essentially getting tenantContractId from the contract's meta (already in itenxxx format but in hex) so we just need to convert the hex to its string representation. However, I mistakenly named the variable tenantContractAddress, which is incorrect - fixed.

File: src/ElvSpace.js
Comment (elv-Noppapon) [inline src/ElvSpace.js:109]: If we do not use the private key of the new account, putMeta will not succeed as it will be blocked by onlyAdmin() on meta_object.sol contract.

File: src/ElvSpace.js
Comment (elv-Noppapon) [inline src/ElvSpace.js:115]: Need to change account or else OnlyAdmin() on meta_object contract will block the following putMeta(s).

File: src/Config.js
Comment (elv-serban) [inline src/Config.js:8]: Let's call this `dev`  (it's more common in blockchain apps)

File: src/ElvAccount.js
Comment (elv-serban) [inline src/ElvAccount.js:34]: I think it is best to remove the check from SetSigner() - because that code could be used for other things, not necessarily fabric tenant operations.

In that case this function can just call Init() and then this.SetAccountTenantContractId (to avoid duplicating the code)

File: src/ElvSpace.js
Comment (elv-serban) [inline src/ElvSpace.js:114]: Here we an use `elvAcount.client`
We already have a client object initialized with the new user account.  
So we can call:
elvAccount.client.SetAccountTenantContractID()
elvAccount.client.CallContractMethodAndWait() ...

File: utilities/EluvioCli.js
Comment (elv-serban) [inline utilities/EluvioCli.js:853]: `Associate this account to the tenant - if an admins group ID is used, it will be converted to the tenant ID.`

File: src/ElvSpace.js
Comment (elv-Noppapon) [reply src/ElvSpace.js:114]: Thanks, I totally missed how elvAccount also has a client!

**Review by elv-serban — APPROVED** [review]

**Pull request merged**
